tests/test_xml.py

```diff
@@ -64,15 +67,42 @@ def test_xml(xmltestcase):
             complaints[currentPartSource].append(
                 {'id': withinid, 'message': formattedMessage, 'type': elemType})
 
-    # Isolate unexpected warnings
+    return complaints
+
+
+def isolatewarnings(complaints):
+    """Isolate warnings
+
+    :param dict complaints:
+    :return: number of unexpected errors or warnings
+    """
+    errors = 0
     for checkmodule, complaintList in complaints.items():
         for complaint in complaintList:
-            if not complaint["id"].count("sdsc.expect.{0}.{1}".format(complaint["type"], checkmodule)):
-                print("Unexpected {0} {1!r} generated by module {2!r} for ID {3!r}.".format(complaint[
-                    "type"], complaint["message"], checkmodule, complaint["id"]), file=sys.stderr)
-                nr_errors += 1
+            expect = complaint["id"].count(
+                "sdsc.expect.{0}.{1}".format(complaint["type"],
+                                             checkmodule))
+            if not expect:
+                print("Unexpected {0} {1!r} "
+                      "generated by module {2!r} "
+                      "for ID {3!r}.".format(complaint["type"],
+                                             complaint["message"],
+                                             checkmodule,
+                                             complaint["id"]),
+                      file=sys.stderr)
+                errors += 1
+    return errors
 
-    # Now check for missing errors and warnings
+
+def searchforerrors(nr_errors, inputids, case, complaints):
+    """Check for missing errors and warnings
```

Comment:
Maybe better: "Checks whether the style checker produces all expected warnings and errors"